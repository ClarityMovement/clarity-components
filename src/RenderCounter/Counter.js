import React from 'react';
import T from 'prop-types';

const styles = {
  counter: {
    display: 'inline-block',
    float: 'right',
    borderRadius: '8px',
    backgroundColor: '#f95',
    fontSize: '8pt',
    fontWeight: 'bold',
    textAlign: 'center',
    padding: '1px 5px',
    color: 'white',
    top: '2px',
  },
};

const Counter = ({ count }) => (
  <div style={styles.counter}>{count}</div>
);

Counter.propTypes = {
  /**
   * Value for Counter to render.
   */
  count: T.number.isRequired,
};

export default Counter;
