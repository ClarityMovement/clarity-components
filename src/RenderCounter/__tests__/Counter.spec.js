/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import ShallowTestRenderer from 'react-test-renderer/shallow';

import Counter from '../Counter';

const shallowRenderer = new ShallowTestRenderer();

describe('Counter({ count }): component', () => {
  it('deeply renders prop count=0', () => {
    const element = renderer.create(<Counter count={0} />);
    expect(element).toMatchSnapshot();
  });

  it('shallowly renders a long prop count', () => {
    const instance = shallowRenderer.render(<Counter count={30002} />);
    expect(instance).toMatchSnapshot();
  });
});

