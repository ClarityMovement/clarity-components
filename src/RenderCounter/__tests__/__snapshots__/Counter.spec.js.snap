// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter({ count }): component deeply renders prop count=0 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#f95",
      "borderRadius": "8px",
      "color": "white",
      "display": "inline-block",
      "float": "right",
      "fontSize": "8pt",
      "fontWeight": "bold",
      "padding": "1px 5px",
      "textAlign": "center",
      "top": "2px",
    }
  }
>
  0
</div>
`;

exports[`Counter({ count }): component shallowly renders a long prop count 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#f95",
      "borderRadius": "8px",
      "color": "white",
      "display": "inline-block",
      "float": "right",
      "fontSize": "8pt",
      "fontWeight": "bold",
      "padding": "1px 5px",
      "textAlign": "center",
      "top": "2px",
    }
  }
>
  30002
</div>
`;
